rule.proto

```diff
@@ -0,0 +1,221 @@
+syntax = "proto2";
+package dv360_excel_plugin;
+
+/* A message that represents a scheduled rule for manipulating DV360 entities.
+ */
+message Rule {
+  /* required
+   * The name of the rule, set by the user.
+   */
+  optional string name = 1;
+
+  /* required
+   * The id that uniquely identified each rule.
+   */
+  optional int32 id = 2;
+
+  // required
+  optional Action action = 3;
+
+  // required
+  optional Condition condition = 4;
+
+  // required
+  optional Schedule schedule = 5;
+}
+
+/*
+ * A message representing the action applied to a DV360 entity.
+ */
+message Action {
+  enum Type {
+    UNSPECIFIED_TYPE = 0;
+    CHANGE_LINE_ITEM_STATUS = 1;
+    CHANGE_BIDDING_STRATEGY = 2;
+    CHANGE_FIXED_BID_PRICE = 3;
+    DUPLICATE_LINE_ITEM = 4;
+  }
+
+  /* required
+   * The type of action.
+   */
+  optional Type type = 1;
+
+  /*
+   * The action parameters to change line items' status.
+   */
+  message ChangeLineItemStatusParams {
+    // Line items to which the action applies.
+    repeated string line_items = 1;
```

Comment:
The DV360 API returns the [lineItemId](https://developers.google.com/display-video/api/reference/rest/v1/advertisers.lineItems#resource:-lineitem) as a string in int64 format, so I just chose string. Should I change it to int64?

Reply:
I remembered I had the exact same discussion earlier with my design as well. My justification was that I'm displaying this as a string in Excel, so I kept the string type in the proto. I would actually recommend keeping this as int64, so that when you parse the LI ids from input box, it will throw error if the string ids contain non numbers.

Reply:
That makes sense!

Reply:
+1 for int, our internal implementation was using int64, the returned json converted it to string.